### PR TITLE
[WIP] Bible : Open on the last page viewed 

### PR DESCRIPTION
**Pull request by nath5394**
⚠️ [Work in progress] ⚠️ 

Expected behavior : when you open the Bible fragment it reloads the last page viewed. 

For the moment it is not working. ❌  This PR will help us to track changes. Any help appreciated 😄 !

### CODE REVIEW

File: app/src/main/java/co/epitre/aelf_lectures/SyncPrefActivity.java [hunk 1]
```diff
@@ -30,6 +30,7 @@
     public static final String KEY_APP_CACHE_MIN_VERSION= "min_cache_version";
     public static final String KEY_APP_CACHE_MIN_DATE = "min_cache_date";
     public static final String KEY_APP_VERSION = "version";
+    public static final String KEY_BIBLE_LAST_PAGE = "";
```
Comment (AMDG2) [inline app/src/main/java/co/epitre/aelf_lectures/SyncPrefActivity.java]: You should put some value here.

File: app/src/main/java/co/epitre/aelf_lectures/SectionBibleFragment.java [hunk 2]
```diff
@@ -124,8 +127,30 @@ public WebResourceResponse shouldInterceptRequest(WebView view, String url) {
...
+        //Save last URL
+        // onPageFinished infos found on https://stackoverflow.com/a/6720004
+        mWebView.setWebViewClient(new WebViewClient() {
+            @Override
+            public void onPageFinished(WebView view, String url) {
+                super.onPageFinished(mWebView, url);
+                String last_url = mWebView.getUrl();
+                Log.d(TAG, "last_url is " + last_url);
+                //TODO: Save the value in a persistent storage, maybe sharedpreferences and use it when the webview is re-created.
+                SharedPreferences settings = getActivity().getPreferences(Context.MODE_PRIVATE);
```
Comment (AMDG2) [inline app/src/main/java/co/epitre/aelf_lectures/SectionBibleFragment.java]: You retrieve a `SharedPreference` using a different method than in line [59](https://github.com/HackMyChurch/aelf-dailyreadings/pull/13/files#diff-c83a481a5a2835d697777ae7a847e12eR59).

So you read from `PreferenceManager.getDefaultSharedPreferences(getContext())` and you write to `getActivity().getPreferences(Context.MODE_PRIVATE)` thus it cannot work.

**Comment by nath5394** [comment]
Thank's to @AMDG2 advices it now works as expected on my phone. Can you review it and in case it is OK merge to masrter ?

File: app/src/main/java/co/epitre/aelf_lectures/SectionBibleFragment.java [hunk 3]
```diff
@@ -124,8 +127,30 @@ public WebResourceResponse shouldInterceptRequest(WebView view, String url) {
...
+                Log.d(TAG, "last_url is " + last_url);
+                SharedPreferences settings = getActivity().getPreferences(Context.MODE_PRIVATE);
+                SharedPreferences.Editor editor = settings.edit();
+                editor.putString(KEY_BIBLE_LAST_PAGE, last_url);
+                editor.apply();
+                Log.d(TAG,"onPageFinished, KEY_BIBLE_LAST_PAGE is " + settings.getString(KEY_BIBLE_LAST_PAGE,"null"));
+
+            }
+        });
+        //TODO: Create the following two methods somewhere ; getLastUrl() returning String ; setLastUrl(String last_url)
```
Comment (yadutaf) [inline app/src/main/java/co/epitre/aelf_lectures/SectionBibleFragment.java]: Can you remove the TODO ?

File: app/src/main/java/co/epitre/aelf_lectures/SyncPrefActivity.java (hunk 1 above)
Comment (yadutaf) [reply app/src/main/java/co/epitre/aelf_lectures/SyncPrefActivity.java]: like "bible_last_page" :)

**Pull request merged**